### DISCUSSION TITLE
能否提供重新修改编辑导出后的书籍中部分译文的功能

### DISCUSSION THREAD

**Discussion by egaojing** [bookfere/Ebook-Translator-Calibre-Plugin#64] [Ideas]
在导出翻译的epub文件之前，都可以修改译文；在导出epub书籍后，现在只能通过sigil这样的软件修改；能否提供再次修改已导出epub书籍的功能，阅读过程中发现翻译不满意的地方，通过插件回到可以修改各个翻译的部分，重新修改部分译文

**Comment by bookfere** [comment]
如果你开启了缓存功能，每次打开同一本书，就会调用缓存中的数据。这意味着，只要没有清空缓存，就可以随时对之前的翻译内容进行修改，然后重新输出电子书。不知这是否能满足你的需求。

**Reply by egaojing** [reply]
您指的修改是通过calibre修改或者类似sigil软件修改，没法通过插件重新修改吧；我意思能重新回到插件翻译过程中，那种逐段对比的状态，然后进行修改

**Reply by bookfere** [reply]
我指的就是用 Ebook Translator 修改，而不是 Calibre 或 Sigil。这里解释一下插件的“缓存”是如何工作的：

开启插件中缓存功能后，用插件打开电子书 A 时，其原文内容都会被存入缓存，翻译期间产生的译文也同样会存入缓存。当翻译完毕导出电子书后，缓存并不会自动消失（在没有手动清空缓存的情况下），下次再次用插件打开电子书 A 时，会自动调用这个缓存，所有内容都会保持上次的状态，因此你可以再次对某些译文进行重新翻译、校正或编辑的操作。

**Reply by egaojing** [reply]
我刚使用插件自动翻译了一本英文书，没有清理过缓存，点开设置-通用-缓存是开启的，并且存在一定体积的缓存，然后（1）点插件打开原英文书籍，左侧是一行一行英文，状态未翻译（2）点插件打开翻译后书籍，左侧是一行英文，一行中文，状态未翻译

**Reply by bookfere** [reply]
使用的插件是当前的最新版本 v2.0.2 吗？

**Reply by egaojing** [reply]
是的

**Reply by bookfere** [reply]
目前缓存有一个唯一的标识，由“文件路径“、”翻译引擎名称“、”目标语言”以及“合并字符数”这四个因素组成，开启缓存功能的状态下，要调用缓存需要同时满足这四个因素才行。

按照你的描述（1）中的步骤，我这里无法重现你所遇到的问题，而描述（2）的做法无意义，因为输出的电子书对插件来说是全新的。

建议这样测试一下：首先清空所有缓存，然后用插件打开一本书，只翻译一行，关闭插件界面，再次打开那本书，确保上述四个因素与上次打开时的一致，看刚才翻译的那一行是否还存在。

**Reply by egaojing** [reply]
谢谢，我试试，不行我就用epub编辑软件修改下，谢谢您开发这样的软件
